docs/_future/listening_responding_functions.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Listening & responding to functions
+lang: en
+slug: functions
+order: 5
+layout: future
+---
+
+<div class="section-content">
+
+Your app can use the `function()` method to listen to incoming function requests. The method requires a function `callback_id` of type `str`. This `callback_id` must also be defined in your [Function](/bolt-python/future/concepts#manifest-functions) definition. Functions must eventually be completed with the `complete()` function to inform Slack that your app has processed the function request. `complete()` requires **one of two** keyword arguments: `outputs` or `error`. There are two ways to complete a Function with `complete()`:
+
+* `outputs` of type `dict` completes your function **successfully** and provide a dictionary containing the outputs of your function as defined in the apps manifest.
+* `error` of type `str` completes your function **unsuccessfully** and provides a message containing information regarding why your function was not successful.
+
+</div>
+
+<div>
+<span class="annotation">Refer to <a href="https://slack.dev/bolt-python/api-docs/slack_bolt/kwargs_injection/args.html" target="_blank">the module document</a> to learn the available listener arguments.</span>
+```python
+# The sample function simply outputs an input
+@app.function("sample_function")
+def sample_func(event, complete: Complete):
+    try:
+        message = event["inputs"]["message"]
+        complete(
+            outputs={
+                "updatedMsg": f":wave: You submitted the following message: \n\n>{message}"
+            }
+        )
+    except Exception as e:
+        complete(error="Cannot submit the message")
+        raise e
+```
+</div>
+
+<details class="secondary-wrapper">
+<summary markdown="0">
+<h4 class="secondary-header">Function Interactivity</h4>
+</summary>
+
+<div class="secondary-content">
+
+The `function()` method returns a `SlackFunction` decorator object. This object can be used by your app to set up interactive listeners such as [actions](/bolt-python/concepts#action-respond) and [views](/bolt-python/concepts#view_submissions). These listener listen to events created during the handling of your `function` event. Additionally they will only be called when a user interacts with a block element that has the following attributes:
+
+* It was created during the handling of a `function` event
```

Comment:
Could we standardize the punctuation for all bullet points in these new docs? It looks like for other existing pages the precedent has been to add a period after each bullet:
```suggestion
* It was created during the handling of a `function` event.
```